src/pykeen/sampling/filtering.py

```diff
@@ -204,10 +204,13 @@ def __init__(self, mapped_triples: MappedTriples):
 
     def contains(self, batch: MappedTriples) -> torch.BoolTensor:  # noqa: D102
         return torch.as_tensor(
-            data=[tuple(triple) in self.triples for triple in batch.tolist()],
+            data=[
+                tuple(triple) in self.triples
+                for triple in batch.view(-1, 3).tolist()
+            ],
             dtype=torch.bool,
             device=batch.device,
-        )
+        ).view(*batch.shape[:-1])
```

Comment:
Should this reshaping happen here, or better outside to avoid ambiguity with incorrect shapes?

Reply:
Being able to pass `(batch_size, num_neg, 3)` is convenient, so I am happy with this change :slightly_smiling_face: